pkg/virtctl/credentials/add-key/add-key.go

```diff
@@ -0,0 +1,280 @@
+	if !cmdFlags.Force {
+		// Only create a secret if VM is not running.
+		_, err := cli.VirtualMachineInstance(vm.Namespace).Get(cmd.Context(), vm.Name, &metav1.GetOptions{})
+		if err == nil {
+			return fmt.Errorf("virtual machine %s is running", vm.Name)
+		}
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("error when getting virtual machine instance: %w", err)
+		}
+	}
+
+	secret := newSecretWithKey(vm, sshKey)
+	secret, err = cli.CoreV1().Secrets(vm.Namespace).Create(cmd.Context(), secret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating secret: %w", err)
+	}
+
+	vmPatched := false
+	defer func() {
+		// If VM patch failed, remove the secret
+		if vmPatched {
+			return
+		}
+		deleteErr := cli.CoreV1().Secrets(vm.Namespace).Delete(cmd.Context(), secret.Name, metav1.DeleteOptions{})
+		if deleteErr != nil && err == nil {
+			// Return the deletion error from the enclosing function
+			err = fmt.Errorf("error deleting secret: %w", deleteErr)
+		}
+	}()
+
+	accessCredential := newAccessCredential(secret.Name, cmdFlags.User)
+	accessCredentialPatch := patchToAddAccessCredential(accessCredential)
+
+	// First, try patch that adds the array
+	fullPatch := common.MustMarshalPatch(append(patchToAddAccessCredentialsArray(), accessCredentialPatch)...)
+	_, err = cli.VirtualMachine(vm.Namespace).Patch(cmd.Context(), vm.Name, types.JSONPatchType, fullPatch, &metav1.PatchOptions{})
+	if err != nil {
+		// If it fails, it probably means that the array exists.
+		// Try to add the new access credential to the existing array.
+		_, err = cli.VirtualMachine(vm.Namespace).Patch(cmd.Context(), vm.Name, types.JSONPatchType, common.MustMarshalPatch(accessCredentialPatch), &metav1.PatchOptions{})
+		if err != nil {
+			return fmt.Errorf("error patching virtual machine: %w", err)
+		}
+	}
+
+	vmPatched = true
+	return nil
+}
+
+func updateSecretWithSshKey(cmd *cobra.Command, cli kubecli.KubevirtClient, cmdFlags *addSshKeyFlags, vm *v1.VirtualMachine, sshKey string) error {
+	secrets := common.GetSshSecretsForUser(vm.Spec.Template.Spec.AccessCredentials, cmdFlags.User)
+	if len(secrets) == 0 {
+		return fmt.Errorf("no secrets specified for user: %s", cmdFlags.User)
+	}
+
+	secretName, err := common.FindSecretOrGetFirst(cmdFlags.Secret, secrets)
+	if err != nil {
+		return err
+	}
+
+	secret, err := cli.CoreV1().Secrets(vm.Namespace).Get(cmd.Context(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error getting secret \"%s\": %w", secretName, err)
+	}
+
+	if secretContainsKey(secret.Data, sshKey) {
+		cmd.Printf("Secret \"%s\" already contains this SSH key.", secretName)
+		return nil
+	}
+
+	if !cmdFlags.Force {
+		// Check if secret is owned by the VM. This is useful to not accidentally update a secret that is used by multiple VMs.
+		var isOwnedByVm bool
```

Comment:
Can you please create a function with these checks and replace the part using it?

Reply:
done